packages/frontend/src/components/MainButton.tsx

```diff
@@ -22,7 +21,14 @@ export const MainButton = (props: {
       }}
       onClick={handler}
     >
-      {children ? children : loading ? <Spinner /> : <p>{message}</p>}
+      {message}
+      {loading && (
+        <>
+          <span className="dot1">.</span>
+          <span className="dot2">.</span>
+          <span className="dot3">.</span>
```

Comment:
we should use a dedicated spinner component for this in the future. I think it's worth doing a rewrite of the frontend with shadcdn components soon